lib/http/2/stream.rb

```diff
@@ -173,16 +173,15 @@ def reprioritize(weight: 16, dependency: 0, exclusive: false)
     # @param payload [String]
     # @param end_stream [Boolean] indicates last response DATA frame
     def data(payload, end_stream: true)
-      flags = []
```

Comment:
Minor thing: can we leave this declaration up here and pass through flags (which is set to empty array)? The benefit is that we don't end up allocating a new object on each iteration of the while loop.